provider/provider.go

```diff
@@ -80,6 +80,7 @@ func New() *schema.Provider {
 			"coder_app":            appResource(),
 			"coder_metadata":       metadataResource(),
 			"coder_script":         scriptResource(),
+			"coder_env":            envResource(),
```

Comment:
Thinking loud: Do we need an "opposite" resource? `coder_no_env` to remove the ENV variable?

Reply:
For sure, I'd say that goes as a prop: `resource "coder_env" "" { name = "FOO", unset = true }`. Discussion ongoing in https://github.com/coder/coder/issues/10166.